src/services/product-service.ts

```diff
@@ -0,0 +1,16 @@
+import Product from "../entities/Product";
+import ProductCategory from "../entities/ProductCategory";
+import { ProductFormData } from "../entities/ProductSchema";
+
+export class ProductService {
+  static generateProduct = (
+    productCategories: ProductCategory[],
+    data: ProductFormData,
+  ): Product => {
+    const category = productCategories.find(
```

Comment:
Before initializing the newProduct, it would be better to check if category is not undefined (better safe than sorry :D)

Reply:
Done.